flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer08.java

```diff
@@ -295,7 +296,7 @@ public void open(Configuration parameters) throws Exception {
 
 	@Override
 	public void run(SourceContext<T> sourceContext) throws Exception {
-		if (fetcher != null) {
+		if (fetcher != null && isAutoCommitEnabled()) {
```

Comment:
Ikes! Added that on the wrong conditional, it was supposed to be on the `!streamingRuntimeContext.isCheckpointingEnabled()` clause where `PeriodicOffsetCommitter` is initialized.